File: evals/behavioral/tasks/typescript/fixtures/total-safe-div.good.ts
// Total: the output is widened to include undefined, so "no answer" (b === 0)
// becomes a case the caller must handle. `safeDiv(1, 0) + 1` does not
// type-check (the operand is possibly undefined).
export function safeDiv(a: number, b: number): number | undefined {
  if (b === 0) return undefined;
  return a / b;
}

File: evals/behavioral/tasks/scala3/fixtures/total-safe-div.good.scala
// Total: the output is widened to Option, so "no answer" (b == 0) becomes a case
// the caller must handle. `safeDiv(1, 0) + 1` does not compile (Option has no +).
def safeDiv(a: Int, b: Int): Option[Int] =
  if b == 0 then None else Some(a / b)
